apps/wallet-dashboard/components/Dialogs/Staking/views/EnterAmountView.tsx

```diff
@@ -1,42 +1,188 @@
 // Copyright (c) 2024 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
-import React from 'react';
-import { Button, Input } from '@/components';
+import React, { useEffect } from 'react';
+import {
+    useFormatCoin,
+    useBalance,
+    CoinFormat,
+    parseAmount,
+    useCoinMetadata,
+    useStakeTxnInfo,
+} from '@iota/core';
+import { IOTA_TYPE_ARG } from '@iota/iota-sdk/utils';
+import {
+    Button,
+    ButtonType,
+    KeyValueInfo,
+    Panel,
+    Divider,
+    Input,
+    InputType,
+    Header,
+    InfoBoxType,
+    InfoBoxStyle,
+    InfoBox,
+} from '@iota/apps-ui-kit';
+import { Field, type FieldProps, useFormikContext } from 'formik';
+import { Exclamation } from '@iota/ui-icons';
+import { useCurrentAccount, useIotaClientQuery } from '@iota/dapp-kit';
 
-interface EnterAmountViewProps {
-    selectedValidator: string | null;
+import { Validator } from './Validator';
+import { StakedInfo } from './StakedInfo';
+import { Layout, LayoutBody, LayoutFooter } from './Layout';
+
+export interface FormValues {
     amount: string;
-    onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
+}
+
+interface EnterAmountViewProps {
+    selectedValidator: string;
+    setAmount: (amount: string) => void;
     onBack: () => void;
     onStake: () => void;
-    isStakeDisabled: boolean;
+    showActiveStatus?: boolean;
+    gasBudget?: string | number | null;
+    handleClose: () => void;
 }
 
 function EnterAmountView({
-    selectedValidator,
-    amount,
-    onChange,
+    selectedValidator: selectedValidatorAddress,
     onBack,
     onStake,
-    isStakeDisabled,
+    gasBudget = 0,
+    handleClose,
+    setAmount,
 }: EnterAmountViewProps): JSX.Element {
+    const coinType = IOTA_TYPE_ARG;
+    const { data: metadata } = useCoinMetadata(coinType);
+    const decimals = metadata?.decimals ?? 0;
+
+    const account = useCurrentAccount();
+    const accountAddress = account?.address;
+
+    const { values } = useFormikContext<FormValues>();
+    const amount = values.amount;
+
+    const { data: system } = useIotaClientQuery('getLatestIotaSystemState');
+    const { data: iotaBalance } = useBalance(accountAddress!);
+    const coinBalance = BigInt(iotaBalance?.totalBalance || 0);
+
+    const maxTokenBalance = coinBalance - BigInt(Number(gasBudget));
+    const [maxTokenFormatted, maxTokenFormattedSymbol] = useFormatCoin(
+        maxTokenBalance,
+        IOTA_TYPE_ARG,
+        CoinFormat.FULL,
+    );
+
+    const _gasBudget = BigInt(gasBudget ?? 0);
```

Comment:
better to not use underscores I think
```suggestion
    const gasBudgetBigInt = BigInt(gasBudget ?? 0);
```